geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/AbstractBatchScan.scala

```diff
@@ -156,13 +155,14 @@ abstract class AbstractBatchScan[T, R <: AnyRef](ranges: Seq[T], threads: Int, b
     */
   private class Terminator extends Runnable {
 
-    private val done = new AtomicBoolean(false)
+    @volatile
+    private var done = false
```

Comment:
OK, why the switch from AtomicBoolean to volatile boolean?

Reply:
i remembered that if you just use .get and .set, there's no difference between the 2, other than the overhead of the AtomicBoolean class...
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/concurrent/atomic/AtomicBoolean.java#L63

Reply:
Good call...  I kinda want about the VarHandle, etc, etc...  I have more questions about how AtomicObjects work now...  but that's for another time.